### PR TITLE
OAS : Ignore anything at PathItem level which is not HTTP verb

### PR DESCRIPTION
**Pull request by philsturgeon**
### Please check if the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features) 
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

For now just trying to write tests that establish the issue. 

### What is the current behavior?

Currently running spectral CLI against [this example](https://gist.githubusercontent.com/jerzyn/d0201558bf52b3659c0d43cac646896c/raw/0bc6e80d7bce9d8556f791a641924beb624e2f7b/swagger-demo.yaml) is throwing a lot of errors:

```
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/products,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/products/{product_id},x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/products/{product_id}/reviews,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/products/{product_id}/reviews-stats,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/users,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/users/{user_id},x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/users/{user_id}/wishlist,x-summary':
TypeError: Cannot convert undefined or null to object
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/users/{user_id}/wishlist/{product_id},x-summary':
```

Related to https://github.com/stoplightio/spectral/issues/109

### CODE REVIEW

**Comment by philsturgeon** [comment]
A better backtrace: 

```
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at exports.oasOp2xxResponse.targetVal (/Users/phil/src/spectral/dist/rulesets/oas/functions/oasOp2xxResponse/index.js:5:30)
    at Object.exports.lintNode (/Users/phil/src/spectral/dist/linter.js:56:31)
    at runRule (/Users/phil/src/spectral/dist/runner.js:47:51)
    at Object.exports.runRules (/Users/phil/src/spectral/dist/runner.js:17:38)
    at Spectral.<anonymous> (/Users/phil/src/spectral/dist/spectral.js:27:29)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/phil/src/spectral/node_modules/tslib/tslib.js:104:62)
Encountered error when running rule 'operation-2xx-response' on node at path '$,paths,/,x-summary':
```

**Comment by P0lip** [comment]
I believe I found the cause of the issue, @philsturgeon.

In `src/rulesets/oas/index.ts` the following query expression `"$..paths.*[?( name() !== 'parameters' )]";` is not quite right 
As you can see, it only filters out parameters, while the spec you added has other fields, e.g. `x-summary`, and therefore Spectral bails out on such properties.
If you remove all `x-summary` properties, it works just fine.
I guess we'd need to change the condition a bit to match actual http operations rather than trying to filter out irrelevant properties.

I guess the condition would need to look as follows
```"$..paths.*[?( name() === 'get' || name === 'head' || name === 'post' )]";```
+ all the other http operations... I hope it's case-insensitive.
Didn't check the above, but hopefully it works.

File: src/rulesets/oas/index.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import { DiagnosticSeverity } from '@stoplight/types';
 import { FunctionCollection, RuleCollection, RuleFunction, RuleType } from '../../types';
 
-export const operationPath = "$..paths.*[?( name() !== 'parameters' )]";
+export const operationPath =
+  "$..paths.*[?( name() === 'get' || name === 'put' || name === 'post'" +
```
Comment (P0lip) [inline src/rulesets/oas/index.ts]:
```suggestion
  "$..paths.*[?( name() === 'get' || name() === 'put' || name() === 'post'
```
+ extra quote
+ I believe we need to call the `name` function

File: src/rulesets/oas/index.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
+  " || name === 'delete' || name === 'options' || name === 'head'" +
```
Comment (P0lip) [inline src/rulesets/oas/index.ts]:
```suggestion
  " || name() === 'delete' || name() === 'options' || name() === 'head' +
```
Ditto

File: src/rulesets/oas/index.ts [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+  " || name === 'patch' || name === 'trace' )]";
```
Comment (P0lip) [inline src/rulesets/oas/index.ts]:
```suggestion
  " || name() === 'patch' || name() === 'trace' )]";
```

**Comment by philsturgeon** [comment]
Ready for review please!

File: src/rulesets/oas/index.ts (hunk 2 above)
Comment (philsturgeon) [reply src/rulesets/oas/index.ts]: Fixed all this and added tests to confirm. Please add thumbs.

**Pull request merged**